product-service/handlers/create-products.ts

```diff
@@ -0,0 +1,66 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import 'source-map-support/register';
+import { Client } from 'pg';
+import headers from './headers';
+import generateDbConfig from '../db/db-config';
+import { addNewProduct } from '../db/queries';
+
+type BodyTypes = {
+  title: string,
+  description: string,
+  count: number,
+  price: number,
+}
+
+const isBodyInvalid = (body: BodyTypes) : boolean => {
+  if (body === null || body === undefined) return true;
+  const { title, description, count, price } = body;
+  return (typeof title !== 'string'
+    && typeof description !== 'string'
+    && typeof count !== 'number'
+    && typeof price !== 'number'
+  );
+}
+
+const createProducts: APIGatewayProxyHandler = async (event, _context) => {
+  console.log('Invoke createProducts lambda\n');
+  console.log('ENVIRONMENT VARIABLES:' + JSON.stringify(process.env, null, 2));
+  console.info('EVENT:' + JSON.stringify(event, null, 2));
+  
+  const body: BodyTypes = JSON.parse(event.body);
+
+  if (isBodyInvalid(body)) {
+    return {
+      statusCode: 400,
+      headers,
+      body: 'Product data is invalid'
+    }
+  } 
+
+  const dbClient = new Client(generateDbConfig(process));
+  await dbClient.connect();
+
+  try {
+    await dbClient.query('BEGIN');
```

Comment:
There's no need it explicit transaction wrapper if you use CTEs, as they are treated like a single statement, i.e. they are run in an implicit transaction. https://dba.stackexchange.com/questions/207459/what-happens-with-with-if-later-the-query-fails